### PR TITLE
fix(social): stabilize follower count across sessions (#2221)

### LINKED ISSUE #2221: follower count
so my follower count keeps changing the count or i mean the number of followers im wondering if that can be fixed its also showing i have about 53 followers on the web version and on the app i have 72 i think 72 or 71 lol but can that be fixed if thats possible?

### PR DESCRIPTION
**Pull request by realmeylisdev**
## Summary

Closes #2221

Follower counts fluctuated across app restarts and showed a mismatch between the web app (~53) and mobile app (~72). Root causes identified:

1. **Web/app mismatch**: Web uses only the Funnelcake REST API count (indexes Divine relay only). Mobile supplements with 3 indexer relays (`purplepag.es`, `user.kindpag.es`, `relay.nos.social`) that see ~19 more followers across the broader Nostr network.
2. **Mobile cross-session fluctuation**: Indexer relay queries use timeouts that return partial results. No persistent cache meant each app restart ran a fresh "relay lottery" yielding different counts.

### Changes

- **Persistent follower count cache** — `SocialService` now stores counts in `SharedPreferences` keyed by pubkey, so counts survive app restarts and serve as a floor for fresh queries
- **Hysteresis logic** — Small drops (<20%) within 1 hour are treated as relay query variance and the higher persisted count is kept. Genuine large drops (>20%) or stale data (>1h) accept the fresh count
- **Network failure fallback** — When all sources return zero, persisted data is returned instead of showing 0
- **Increased indexer timeout** — 5s → 8s in both `SocialService` and `FollowRepository` to reduce partial-result variance
- **10 unit tests** covering persistence, hysteresis thresholds, boundary conditions, stale data, network failure, and in-memory caching

### Files changed

| File | Change |
|------|--------|
| `mobile/lib/services/social_service.dart` | Persistent cache, hysteresis, increased timeout |
| `mobile/lib/repositories/follow_repository.dart` | Increased indexer timeout 5s → 8s |
| `mobile/lib/providers/app_providers.dart` | Pass `SharedPreferences` to `SocialService` |
| `mobile/lib/providers/app_providers.g.dart` | Codegen update |
| `mobile/test/services/social_service_follower_stats_test.dart` | New: 10 unit tests |

> **Note**: The web/app count gap is a separate concern requiring changes in `divine-web` (supplement Funnelcake count with indexer relay queries) or server-side aggregation in Funnelcake. This PR stabilizes the mobile side.

## Test plan

- [x] `flutter test test/services/social_service_follower_stats_test.dart` — 10/10 pass
- [x] `flutter test test/repositories/follow_repository_test.dart` — 70/70 pass (no regression)
- [x] `flutter analyze` — no issues
- [ ] Manual: open profile, note follower count, kill app, reopen — count should remain stable
- [ ] Manual: compare count with web — mobile may show higher (expected, due to broader relay coverage)

### CODE REVIEW

File: mobile/lib/services/social_service.dart [hunk 1]
```diff
@@ -193,10 +307,58 @@ class SocialService {
...
       );
+
+      // On network failure, try returning persisted data so the UI
+      // shows the last known count instead of zero.
+      final persisted = _loadPersistedStats(pubkey);
+      if (persisted != null) {
+        final fallback = {
+          'followers': persisted.followers,
+          'following': persisted.following,
+        };
```
Comment (hm21) [inline mobile/lib/services/social_service.dart]: Could we create a private class in that file that includes followers and following to improve type safety?

File: mobile/lib/services/social_service.dart
Comment (omartinma) [inline mobile/lib/services/social_service.dart:1]: Since we have a dedicated repository for follow operations, FollowRepository, could we move completely the logic there? Also a more robust cache approach instead of SharedPreferences would be worth to implement

File: mobile/lib/services/social_service.dart (hunk 1 above)
Comment (realmeylisdev) [reply mobile/lib/services/social_service.dart]: Already done — \`FollowerStats\` at \`follow_repository.dart:38-59\` bundles both counters with \`Equatable\`-style equality:

```dart
@immutable
class FollowerStats {
  const FollowerStats({required this.followers, required this.following});
  final int followers;
  final int following;
  static const zero = FollowerStats(followers: 0, following: 0);
}
```

This was introduced when the stats logic moved to \`FollowRepository\` in 63a8644f3.

File: mobile/lib/services/social_service.dart
Comment (realmeylisdev) [reply mobile/lib/services/social_service.dart:1]: Both addressed in the latest commits:
- All follower stats logic moved to \`FollowRepository\` in 63a8644f3
- SharedPreferences replaced with Drift persistence via \`ProfileStatsDao\` (robust, typed, with \`upsertStats\`/\`getStatsRaw\` methods)

Comment (omartinma) [inline mobile/lib/services/social_service.dart:1]: Not related to this, but it might be worth to do a follow up to continue removing things from this service in favor of different repositories

**Review by omartinma — APPROVED** [review]

**Review by hm21 — APPROVED** [review]
